### PR TITLE
feat: Update ASC functional tests to use new aes256 Encrypt algorithm

### LINKED ISSUE #541: Update ASC functional tests to use new `aes256` Encrypt algorithm
The `aes` algorithm has been deprecated and replaced with `aes256` 

The `functional-tests`  profile has been update to specify the  `aes256`  algorithm.

Resulting bytes are as follows:

| initialization vector | ciphertext | signing hash |
| ---------- | ---------- | ---- |
| 16 bytes   | variable bytes | 32 bytes |

**Steps to decrypt:**

1. Base64 decode the response
2. Extract the 16 bytes of Init Vector from the beginning of the data
3. Extract the 32 bytes of the Hash signature from the end of the data
4. Extract the cipher text (remaining bytes in the middle)
5. Perform AES256 decode with the following data/settings
    - Key (first 32 bytes of the 64 byte key used by the encrypt function
    - IV from extracted above
    - 256bits
    - CBC
    - PKCS7
    - cipher text extracted above

**Steps to verify the HASH signature** 

1. Extract the 32 bytes of the Hash signature from the end of the data
2. Run SHA512 hash on the remaining data (IV &  ciphertext) using the the second half of the 64 byte key 
3. Verify extract hash matches calculated hash

### PR DESCRIPTION
**Pull request by cherrycl**
Fix #541 implement decrypt validation

Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by cherrycl** [comment]
Hash validation is added on this PR. Fix #696

**Comment by bnevis-i** [comment]
In order to provide a good example to users, we need to do like is done in https://github.com/codahale/etm/blob/master/etm.go#L204

That is, don't separate verifying the MAC from decrypting the data.  Must verify the MAC, THEN decrypt.

By separating the two operations, someone reading the test might think that only decrypt is needed, which might allow an attacker to recover the key by playing phony ciphertexts at at potential decryptor.

**Comment by cloudxxx8** [comment]
@bnevis-i they will be implemented in the same test.  we just separated this task to two issues (phases), because we need more time to investigate how to test the hash.

**Comment by cloudxxx8** [comment]
@bnevis-i all the tests are implemented in this PR although there are two issues. could you please help review it?

**Review by bnevis-i — DISMISSED** [review]
Approved with strong reservations about splitting hash verification and decryption into two separable operations.

**Comment by cloudxxx8** [comment]
@bnevis-i , ok , looks like I misunderstood your concern about separating the two operations. I got it now. I will ask @cherrycl to modify it.

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: TAF/utils/src/setup/aes256-decrypt.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    else:
+        return "false", extract_hash, calculated_hash
+
+
+if method == "decrypt":
+    decrypted_value = decrypt(hex_value, key)
+    print(decrypted_value)
+elif method == "hash":
+    compare_result = hash(hex_value, key)
+    print(compare_result)
```
[the remark this answers is not in the document]
Comment (cherrycl) [reply TAF/utils/src/setup/aes256-decrypt.py]: Updated.

File: TAF/testScenarios/functionalTest/V2-API/app-service/trigger/POST-positive.robot [hunk 2]
```diff
@@ -57,17 +57,17 @@ TriggerPOST005 - Trigger pipeline (JSON-ZLIB)
...
+    Given Store Secret Data With AES256 Auth
+    And Set Functions FilterByDeviceName, Encrypt, SetResponseData
+    And Set Encrypt Algorithm aes256
     When Trigger Function Pipeline With Matching DeviceName
     Then Should Return Status Code "200"
     And Should Return Content-Type "text/plain"
-    And Body Should Match JSON-ZLIB-AES String
     And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And Decrypted Response Should be Correct
+    And Extract HASH Matches Calculated HASH
```
[the remark this answers is not in the document]
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/app-service/trigger/POST-positive.robot]: Updated.

**Comment by cherrycl** [comment]
@bnevis-i @cloudxxx8 PR is updated, please help to review again. Thank you.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by cherrycl** [comment]
Rebased PR.

**Review by bnevis-i — APPROVED** [review]
LGTM

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**